src/Microsoft.AspNet.TestHost/HttpResponseMessageExtensions.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.TestHost
+{
+    public static class HttpResponseMessageExtensions
+    {
+        public static Task PipelineCompleteAsync(this HttpResponseMessage message)
```

Comment:
I think this is more complicated than we want the test client to be. We want HttpClient to be used like it normally would. If you really need to control the server data structures and flow then you can call TestServer.Invoke directly.